homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -0,0 +1,164 @@
+
+from .const import CONF_BDADDR, CONF_NAME, DEFAULT_RETRY_COUNT, DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+class MicroBotConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for MicroBot."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+        self._discovered_adv: MicroBotAdvertisement | None = None
+        self._discovered_advs: dict[str, MicroBotAdvertisement] = {}
+        self._client: Any | None
+        self._ble_device: BLEDevice | None
+        self._name = None
+        self._bdaddr: str
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug("Discovered bluetooth device: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        parsed = parse_advertisement_data(
+            discovery_info.device, discovery_info.advertisement
+        )
+        self._discovered_adv = parsed
+        data = parsed.data
+        self.context["title_placeholders"] = {
+            "name": data["local_name"],
+            "address": discovery_info.address,
+        }
+        return await self.async_step_init()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+        else:
+            current_addresses = self._async_current_ids()
+            for discovery_info in async_discovered_service_info(self.hass):
+                address = discovery_info.address
+                if (
+                    address in current_addresses
+                    or address in self._discovered_advs
+                ):
+                    continue
+                parsed = parse_advertisement_data(
+                    discovery_info.device, discovery_info.advertisement
+                )
+                if parsed:
+                    self._discovered_advs[address] = parsed
+
+        if not self._discovered_advs:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_BDADDR): vol.In(
+                    {
+                        address: f"{parsed.data['local_name']} ({address})"
+                        for address, parsed in self._discovered_advs.items()
+                    }
+                ),
+                vol.Required(CONF_NAME): str,
```

Comment:
Usually we don't ask for a name in new integrations as the can rename in the UI.

You can also use the title for the name and derive the name from that. There is an example in the latest switchbot update as well as flux_led and wiz